### PR TITLE
Upgrade Cassandra to 4.1.3 and DataStax driver for Cassandra to 4.17.0

### PR DESCRIPTION
**Pull request by asolimando**
See https://issues.apache.org/jira/browse/CALCITE-6243 for all details

### CODE REVIEW

File: cassandra/src/test/java/org/apache/calcite/test/CassandraExtension.java [hunk 1]
```diff
@@ -175,14 +189,6 @@ private CassandraResource() {
         Thread.currentThread().interrupt();
       }
       Stage.shutdownNow();
-
-      if (FBUtilities.isWindows) {
```
Comment (JiajunBernoulli) [inline cassandra/src/test/java/org/apache/calcite/test/CassandraExtension.java:179]: Is this because Cassandra dropped Windows?

Comment (asolimando) [reply cassandra/src/test/java/org/apache/calcite/test/CassandraExtension.java:179]: That's correct, those were Windows-specific methods that C* dropped when they stopped supporting it

**Comment by asolimando** [comment]
@JiajunBernoulli, do you have any other comments for this PR?

I plan to merge it sometime this weekend if there are no concerns.

**Comment by JiajunBernoulli** [comment]
> @JiajunBernoulli, do you have any other comments for this PR?
> 
> I plan to merge it sometime this weekend if there are no concerns.

No, please squash them before merging.

**Pull request merged**